extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx

```diff
@@ -78,12 +79,18 @@ export const LedgerSign = () => {
         }),
       );
       if (signWithLedger.fulfilled.match(res)) {
-        dispatch(
-          submitFreighterTransaction({
-            signedXDR: res.payload,
-            networkDetails,
-          }),
-        );
+        if (shouldSubmit) {
+          dispatch(
+            submitFreighterTransaction({
+              signedXDR: res.payload,
+              networkDetails,
+            }),
+          );
+        } else {
+          // right now there are only two cases after signing,
+          // submitting to network or handling in background script
+          await handleSignedHwTransaction({ transactionXDR: res.payload });
```

Comment:
this is so so pedantic, but I'm wondering if it might be nice to call this param `signedXDR` instead of just `transactionXDR`, similar to the `submitFreighterTransaction` param above.

It would also be nice in `handleSignedHwTransaction` to make it very clear that Ledger has already signed the XDR that is being passed

Reply:
ooh good catch I agree. I think I'll change to `signedTransaction` since we're using that in the background script popupMessageListener already and to keep all the `handleSignedHwTransaction` methods using the same var name

Reply:
https://github.com/stellar/freighter/pull/573/commits/129cfb6529859ded178012074482d3043e83960d